### PR TITLE
Some fix on definitions

### PR DESCRIPTION
**Pull request by nyaoouo**

### CODE REVIEW

File: SaintCoinach/Definitions/Materia.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
   "definitions": [
     {
       "type": "repeat",
-      "count": 10,
+      "count": 16,
```
Comment (Supamiu) [inline SaintCoinach/Definitions/Materia.json:7]: While I understand your reasoning behind that, it's full of 0 atm and if there's a rule we are following on definitions, it's to never assume what will be added in the future. So assuming these columns are for 6 more materia grades is not the way to go until we have a proof for that.

File: SaintCoinach/Definitions/SpearfishingItem.json [hunk 2]
```diff
@@ -37,6 +37,10 @@
         "target": "TerritoryType"
       }
     },
+    {
+      "index": 6,
+      "name": "folklore"
```
Comment (Supamiu) [inline SaintCoinach/Definitions/SpearfishingItem.json:42]: This doesn't follow naming convention and it looks like it's wrong, this column is always 0 so we can't assume what it's supposed to be.

File: SaintCoinach/Definitions/Stain.json [hunk 3]
```diff
@@ -17,7 +17,7 @@
       "name": "SubOrder"
     },
     {
-      "index": 3,
+      "index": 4,
```
Comment (Supamiu) [inline SaintCoinach/Definitions/Stain.json:20]: Both 3 and 4 are using the same value, it's both name, the only difference seems to be with second one being maybe for a different usage, as it's sometimes different in other languages.

Anyways, 3 is still Name regardless.

**Review by Supamiu — CHANGES_REQUESTED** [review]
Hi, thank you for contributing to SaintCoinach.

This PR comes with no context, 6.2 references in commit messages that are up to 8 months old, doesn't follow naming conventions and contains half the changes that are wrong, please make sure to add context to the PR using the PR's message and make sure your PR contains updates that are valid, because checking every single edit is really not fun.

File: SaintCoinach/Definitions/DeepDungeon.json [hunk 4]
```diff
@@ -30,9 +30,9 @@
       }
     },
     {
-      "index": 18,
+      "index": 19,
```
[the remark this answers is not in the document]
Comment (nyaoouo) [reply SaintCoinach/Definitions/DeepDungeon.json:33]: for Deep Dungeon sheet
col 18: offset 28
col 19: offset 16
col 20: offset 17
col 21: offset 18
col 18 is used in 
ffxiv_dx11.exe ver. 2023.07.26.0000.0000: 140A637E6
image
which is not a list of data but a type enum

File: SaintCoinach/Definitions/Materia.json (hunk 1 above)
Comment (nyaoouo) [reply SaintCoinach/Definitions/Materia.json:7]: ffxiv_dx11.exe ver. 2023.07.26.0000.0000: 140988E10
image
a size 16 iterator is used

File: SaintCoinach/Definitions/Stain.json (hunk 3 above)
Comment (nyaoouo) [reply SaintCoinach/Definitions/Stain.json:20]: ffxiv_dx11.exe ver. 2023.07.26.0000.0000: 141201F90
col 3: offset 0
col 4: offset 4
image

ffxiv_dx11.exe ver. 2023.03.24.0000.0000: 14121D200
col 3: offset 0
image

Comment (nyaoouo) [reply SaintCoinach/Definitions/Stain.json:20]: and i dont find any reference in new code keep using offset 0

File: SaintCoinach/Definitions/DeepDungeon.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (Supamiu) [reply SaintCoinach/Definitions/DeepDungeon.json:33]: Then what's col 18? Did you check against ingame data as you're in deep dungeon?

[the remark this answers is not in the document]
Comment (Supamiu) [reply SaintCoinach/Definitions/DeepDungeon.json:33]: I'm not really doing exe digging to update definitions, it's just based on what it's matching ingame as we compare data and UI.

[the remark this answers is not in the document]
Comment (Supamiu) [reply SaintCoinach/Definitions/DeepDungeon.json:33]: It means that you should implement it as a complexlink for col 18 then, better do it in this PR.

Here is an example of a complexlink: https://github.com/xivapi/SaintCoinach/blob/2f894d73e690dbe1775a9edf767d74ea5bd6b422/SaintCoinach/Definitions/Achievement.json#L63

**Pull request closed without merging**